webservices/cxf/src/java/org/sakaiproject/webservices/TestsAndQuizzes.java

```diff
@@ -330,4 +334,92 @@ private String findAssessmentTemplateId(String title) {
 		return null;
 	}
 	
+	/** 
+	 * userPoolsAttachmentReport - Makes a report of all attachments included in all question pools of a specified user. Optionally, fix the broken attachments making a copy of them in a new context.
+	 *
+	 * @param	String sessionId			the id of a valid session for user owner of the pool (NOT admin user)
+	 * @param	String user					the user EID we want to look into their question pools
+	 * @param	String contextToReplace		a site ID where user has access to, so broken attachments will be copied there and replaced in pool  
+	 * @return	String	       		 		a report of the attachments at every pool of the current user and the actions done on them 
+	 * 
+	 */
+	public String userPoolsAttachmentReport(String sessionId, String user, String contextToReplace)
+	{
+		LOG.debug("WS TestsAndQuizzes.userPoolsAttachmentReport(): user - " + user);
+		
+		String userId=null;
+		try
+		{
+			userId=userDirectoryService.getUserId(user);
+		}
+		catch (Exception e)
+		{
+			LOG.warn("WS getUserId() failed for user: " + user);
+			return "";
+		}
+
+		QuestionPoolService qpService = new QuestionPoolService();
+		StringBuilder resultado = new StringBuilder();
+
+		if (qpService != null)
+		{
+			Long miPoolId = null;
+
+			//Get pools (empty list of ids) for a specific user
+			QuestionPoolIteratorFacade qpif = (QuestionPoolIteratorFacade) qpService.getAllPools(userId);
+			try
+			{
+				while (qpif.hasNext())
+				{
+					QuestionPoolFacade qp = (QuestionPoolFacade) qpif.next();
+					resultado.append(this.poolAttachmentReport(sessionId, user, qp.getQuestionPoolId(), contextToReplace));
```

Comment:
Do you mean calling poolAttachmentReport from userPoolsAttachmentReport? It was just for not repeating code. The first one looks for attachments in one single pool. The second one search in all pools related to user, so is calling the first for each pool. How do you suggest combining them?